burn-tensor/src/tensor/api/base.rs

```diff
@@ -530,6 +530,184 @@ where
             .map(|v| Self::new(v))
             .collect()
     }
+
+    /// Pads the tensor according to the specified padding configuration and mode.
+    ///
+    /// This method pads the last two dimensions of a Float or Int tensor with a given padding size and value, according to the specified padding mode.
+    /// Currently, only constant padding mode is implemented, which pads the tensor with a constant value.
+    ///
+    /// # Arguments
+    ///
+    /// * `pad` - A `Padding` struct specifying the size of padding in each dimension.
+    /// * `pad_mode` - The padding mode, determining how the padding is applied. Currently, only `PadMode::Constant` is supported.
+    /// * `value` - The value to be used for padding in constant mode.
+    ///
+    /// # Returns
+    ///
+    /// A new `Tensor<B, D, K>` instance with the specified padding applied.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use burn_tensor::backend::Backend;
+    /// use burn_tensor::{Data,PadMode,Padding,Shape,Tensor};
+    ///
+    /// fn pad_example<B: Backend>()
+    /// where
+    ///     B::FloatElem: From<f32>, // Ensure that FloatElem can be created from a floating-point number
+    /// {
+    ///    let pad_const = 1.1;
+    ///    let tensor = Tensor::<B, 4>::ones_devauto(Shape::new([2, 3, 4, 5]));
+    ///    let padding = Padding::uniform(2);
+    ///
+    ///    let padded_tensor = tensor.pad(padding, PadMode::Constant, Some(pad_const.into()));
+    /// }    
+    ///
+    ///
+    /// ```
+    /// Pads the tensor with the specified padding configuration and padding mode.
+    pub fn pad(
+        self,
+        pad_amount: Padding,
+        pad_mode: PadMode,
```

Comment:
I would change a bit the API

`Padding` => `PadSize`.

The mode should contains the value:

```rust
let tensor = tensor.pad(PadSize::uniform(2), PadMode::Constant(0));
```